app/src/main/java/com/example/bigowlapp/utils/ScheduledLocationTrackingManager.java

```diff
@@ -0,0 +1,81 @@
+package com.example.bigowlapp.utils;
+
+import android.Manifest;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+
+import androidx.core.content.ContextCompat;
+
+import com.example.bigowlapp.model.Schedule;
+import com.example.bigowlapp.service.LocationBroadcastReceiver;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.Geofence;
+import com.google.android.gms.location.GeofencingClient;
+import com.google.android.gms.location.GeofencingRequest;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.GeoPoint;
+
+public class ScheduledLocationTrackingManager {
+
+    private static final int DEFAULT_TRACKING_RADIUS_METERS = 300;
+    private static final long DEFAULT_TRACKING_TIME_MILLISECONDS = 30L * Constants.MINUTE_TO_MILLISECONDS;
+    // use 0 for instant response, and allow delay for better battery life
+    private static final int DEFAULT_MAX_NOTIFY_DELAY_MILLISECONDS = 5 * Constants.MINUTE_TO_MILLISECONDS;
```

Comment:
I testing your code, and, for some reason, the geofence feature does not notify/check every 5 minutes if a user is not at the desired location.

Although, I'll give you an interesting find: it only re-notified when I opened the Google Maps app, so I guess you have to "wake" up the location feature.

Reply:
I've noticed that too. Will try to use a recurring alarm to fix this issue.

Reply:
Seems like the recurring alarm idea worked.